[docs] Improve the SvgIcon documentation (#18560)
mui-org/material-ui: docs/src/pages/components/icons/icons.md

## Material Icons

> Material Design has standardized over 1,000 official icons, each in five different "themes" (see below). For each SVG icon, we export the respective React component from the @material-ui/icons package. You can search the full list of these icons in our [built-in search page](/components/material-icons/).

Reviewer: ```suggestion
Material Design has standardized over 1,000 official icons, each in five different "themes" (see below). For each SVG icon, we export the respective React component from the @material-ui/icons package. You can [search the full list of these icons](/components/material-icons/).
```

### Usage

> Install @material-ui/icons. Import icons using one of these two options:

Reviewer: ```suggestion
Install `@material-ui/icons`. Import icons using one of these two options:
```

Make sure you follow the [minimizing bundle size guide](/guides/minimizing-bundle-size/#option-2) before using the second approach.
The configuration of a Babel plugin is encouraged.

> Each icon also has a theme: Filled (default), Outlined, Rounded, Two tone and Sharp. If you want to import the icon component with a theme different than default, append the theme name to the icon name. For example `@material-ui/icons/Delete` icon with:

Reviewer: ```suggestion
Each icon also has a "theme": Filled (default), Outlined, Rounded, Two tone and Sharp. If you want to import the icon component with a theme different than default, append the theme name to the icon name. For example `@material-ui/icons/Delete` icon with:
```

> Each icon also has a "theme": Filled (default), Outlined, Rounded, Two tone and Sharp. If you want to import the icon component with a theme different than default, append the theme name to the icon name. For example `@material-ui/icons/Delete` icon with:

Reviewer: ```suggestion
Each icon also has a "theme": Filled (default), Outlined, Rounded, Two tone and Sharp. If you want to import the icon component with a theme other than default, append the theme name to the icon name. For example `@material-ui/icons/Delete` icon with:
```

- Filled theme (default) is exported as `@material-ui/icons/Delete`,
- Outlined theme is exported as `@material-ui/icons/DeleteOutlined`,
- Rounded theme is exported as `@material-ui/icons/DeleteRounded`,
- Twotone theme is exported as `@material-ui/icons/DeleteTwoTone`,
- Sharp theme is exported as `@material-ui/icons/DeleteSharp`.

> Note: The Material Design specification names the icons using "snake_case" naming (for example `delete_forever`, `add_a_photo`), while @material-ui/icons exports the respective icons using "PascalCase" naming (for example `DeleteForever`, `AddAPhoto`). There are three exceptions to this naming rule: `3d_rotation` exported as `ThreeDRotation`, `4k` exported as `FourK`, and `360` exported as `ThreeSixty`.

Reviewer: ```suggestion
> Note: The Material Design specification names the icons using "snake_case" naming (for example `delete_forever`, `add_a_photo`), while `@material-ui/icons` exports the respective icons using "PascalCase" naming (for example `DeleteForever`, `AddAPhoto`). There are three exceptions to this naming rule: `3d_rotation` exported as `ThreeDRotation`, `4k` exported as `FourK`, and `360` exported as `ThreeSixty`.
```

{{"demo": "pages/components/icons/SvgMaterialIcons.js"}}

## SvgIcon

If you need a custom SVG icon (not available in the Material Icons [default set](/components/material-icons/)) you can use the `SvgIcon` wrapper.

> The component extends the native `<svg>` element:

Reviewer: ```suggestion
This component extends the native `<svg>` element:
```

- It comes with built-in accessibility.

> - SVG elements should be scaled for a 24x24px viewport, so the resulting icon can be used as is, or included as a child for other Material-UI components that use icons (can be customized with the `viewBox` attribute).

Reviewer: ```suggestion
- SVG elements should be scaled for a 24x24px viewport, so the resulting icon can be used as is, or included as a child for other Material-UI components that use icons. (This can be customized with the `viewBox` attribute).
```

Make sure you follow the [minimizing bundle size guide](/guides/minimizing-bundle-size/#option-2) before using the second approach.
The configuration of a Babel plugin is encouraged.

> Each icon also has a "theme": Filled (default), Outlined, Rounded, Two tone and Sharp. If you want to import the icon component with a theme different than default, append the theme name to the icon name. For example `@material-ui/icons/Delete` icon with:

Reviewer: Thanks. (It distinguishes it from the MUI theme).

> Each icon also has a "theme": `Filled` (default), `Outlined`, `Rounded`, `Two tone` and `Sharp`. If you want to import the icon component with a "theme" different than default, append the "theme" name to the icon name. For example `@material-ui/icons/Delete` icon with:

Reviewer: Could possibly remove the back-ticks around the theme names (or italicize them instead).

Author: I have tried to keep the backtick around `@material-ui/icons/X`. It seems to be acceptable now, as we have removed the others. I feel that the page can accept a small number of "backticked" before it gets overwhelming. What do you think of the output with the latest commit? Should we remove it here too?

Reviewer: > I have tried to keep the backtick around @material-ui/icons/X

That doesn't seem to be the case. See my suggestion for what I meant.

> - `Filled` "theme" (default) is exported as *@material-ui/icons/Delete*,

Reviewer: ```suggestion
- Filled theme (default) is exported as `@material-ui/icons/Delete`,
```

etc.

- `Outlined` "theme" is exported as *@material-ui/icons/DeleteOutlined*,
- `Rounded` "theme" is exported as *@material-ui/icons/DeleteRounded*,
- `Two tone` "theme" is exported as *@material-ui/icons/DeleteTwoTone*,

> - `Sharp` "theme" is exported as *@material-ui/icons/DeleteSharp*.

Reviewer: If anything, we should remove the back-ticks around the theme name, and the quotes from around theme (we've already established in the previous paragraph that "theme" is oddly named). I din't think we should remove the backticks from the export / import.

Note: The Material Design specification names the icons using "snake_case" naming (for example `delete_forever`, `add_a_photo`), while `@material-ui/icons` exports the respective icons using "PascalCase" naming (for example `DeleteForever`, `AddAPhoto`). There are three exceptions to this naming rule: `3d_rotation` exported as `ThreeDRotation`, `4k` exported as `FourK`, and `360` exported as `ThreeSixty`.

{{"demo": "pages/components/icons/SvgMaterialIcons.js"}}

## SvgIcon

If you need a custom SVG icon (not available in the Material Icons [default set](/components/material-icons/)) you can use the `SvgIcon` wrapper.
The component extends the native `<svg>` element:

- It comes with built-in accessibility.
- SVG elements should be scaled for a 24x24px viewport, so the resulting icon can be used as is, or included as a child for other Material-UI components that use icons (can be customized with the `viewBox` attribute).
- By default, the component inherits the current color.

> Optionally, you can set the icon color using one of the theme color prop.

Reviewer: ```suggestion
Optionally, you can apply one of the theme colors using the `coior` prop.
```
?

```jsx
function HomeIcon(props) {
  return (
    <SvgIcon {...props}>
      <path d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z" />
    </SvgIcon>
  );
}
```

### Color

{{"demo": "pages/components/icons/SvgIconsColor.js"}}

### Size

{{"demo": "pages/components/icons/SvgIconsSize.js"}}

> ### Component prop

Author: The controversial part.

Reviewer: Controversial?

Author: There is an ongoing discussion in #17287 regarding removing the support for this prop.
